pyanaconda/ui/tui/spokes/storage.py

```diff
@@ -323,42 +315,37 @@ def input(self, args, key):
             else:
                 return super(StorageSpoke, self).input(args, key)
 
-    def run_dasdfmt(self, to_format):
-        """
-        This generates the list of DASDs requiring dasdfmt and runs dasdfmt
-        against them.
-        """
-        # if the storage thread is running, wait on it to complete before taking
-        # any further actions on devices; most likely to occur if user has
-        # zerombr in their ks file
-        threadMgr.wait(THREAD_STORAGE)
+    def run_dasdfmt_dialog(self, dasd_formatting):
+        """Do DASD formatting if user agrees."""
+        # Prepare text of the dialog.
+        text = ""
+        text += _("The following unformatted or LDL DASDs have been "
+                  "detected on your system. You can choose to format them "
+                  "now with dasdfmt or cancel to leave them unformatted. "
+                  "Unformatted DASDs cannot be used during installation.\n\n")
+
+        text += dasd_formatting.dasds_summary + "\n\n"
+
+        text += _("Warning: All storage changes made using the installer will "
+                  "be lost when you choose to format.\n\nProceed to run dasdfmt?\n")
+
+        # Run the dialog.
+        question_window = YesNoDialog(text)
+        ScreenHandler.push_screen_modal(question_window)
+        if not question_window.answer:
+            return None
+
+        print(_("This may take a moment."), flush=True)
+
+        # Do the DASD formatting.
+        dasd_formatting.report.connect(self._show_dasdfmt_report)
+        dasd_formatting.run(self.storage, self.data)
+        dasd_formatting.report.disconnect(self._show_dasdfmt_report)
```

Comment:
Shouldn't you wait for the Storage thread to finish before calling these methods?

Reply:
Fixed in the input method.